src/SFA.DAS.EmployerCommitmentsV2.Web/Controllers/DraftApprenticeshipController.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using MediatR;
+using Microsoft.AspNetCore.Mvc;
+using SFA.DAS.Apprenticeships.Api.Client;
+using SFA.DAS.Apprenticeships.Api.Types;
+using SFA.DAS.Commitments.Shared.Extensions;
+using SFA.DAS.Commitments.Shared.Interfaces;
+using SFA.DAS.Commitments.Shared.Models;
+using SFA.DAS.CommitmentsV2.Api.Types.Requests;
+using SFA.DAS.CommitmentsV2.Api.Types.Validation;
+using SFA.DAS.EmployerCommitmentsV2.Extensions;
+using SFA.DAS.EmployerCommitmentsV2.Web.Models;
+using SFA.DAS.EmployerCommitmentsV2.Web.Requests;
+using SFA.DAS.ProviderUrlHelper;
+
+namespace SFA.DAS.ProviderCommitments.Web.Controllers
+{
+    
+    [Route("accounts/{hashedAccountId}/apprentices/")]
+    public class DraftApprenticeshipController : Controller
+    {
+        private readonly ICommitmentsService _employerCommitmentsService;
+        private readonly IMapper<AddDraftApprenticeshipViewModel, AddDraftApprenticeshipRequest> _addDraftApprenticeshipToCohortRequestMapper;
+        private readonly ILinkGenerator _urlHelper;
+        private readonly ITrainingProgrammeApiClient _trainingProgrammeApiClient;
+
+        public DraftApprenticeshipController(
+            ICommitmentsService employerCommitmentsService,
+            IMapper<AddDraftApprenticeshipViewModel, AddDraftApprenticeshipRequest> addDraftApprenticeshipToCohortRequestMapper,
+            ILinkGenerator urlHelper,
+            ITrainingProgrammeApiClient trainingProgrammeApiClient)
+        {
+            _employerCommitmentsService = employerCommitmentsService;
+            _addDraftApprenticeshipToCohortRequestMapper = addDraftApprenticeshipToCohortRequestMapper;
+            _urlHelper = urlHelper;
+            _trainingProgrammeApiClient = trainingProgrammeApiClient;
+        }
+
+        [HttpGet]
+        [Route("add")]
+        public async Task<IActionResult> AddDraftApprenticeship(ReservationsAddDraftApprenticeshipRequest request)
```

Comment:
ReservationsAddDraftApprenticeshipRequest  doesn't implement the IAuthorization interface so cohortId will not be populated unless it's passed in (which my be ok at the moment)

Reply:
Believe you meant IAuthorizationContextModel - have added. No idea what this interface does - it has no methods, which presumably means it's a marker interface. I wonder what Uncle Bob would say.